src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -832,6 +832,12 @@ public override XmlNodeType MoveToContent()
                 {
                     if (_node.NodeType != XmlNodeType.Text && _node.NodeType != XmlNodeType.CDATA)
                         break;
+
+                    if (_trailByteCount > 0)
+                    {
+                        break;
+                    }
+
```

Comment:
The ndp\fx\src\cdf is the SL code base. It is also what current GH code started with. I suspect the code change discussed here was added to the desktop framework code base after the SL fork.